packages/design-system/src/components/combobox.tsx

```diff
@@ -168,6 +170,14 @@ export const useCombobox = <Item,>({
   const downshiftProps = useDownshiftCombobox({
     ...rest,
     items: filteredItems,
+    isOpen,
+    onIsOpenChange(value) {
+      // Don't set isOpen to true if there are no items to show
+      // because otherwise first ESC press will try to close it and only next ESC
+      // will reset the value. When list is empty, first ESC should reset the value.
+      const nextIsOpen = value.isOpen === true && filteredItems.length !== 0;
+      setIsOpen(nextIsOpen);
+    },
     defaultHighlightedIndex: -1,
     selectedItem: selectedItem ?? null, // Prevent downshift warning about switching controlled mode
     stateReducer,
```

Comment:
please fix few lines below `inputValue: value ? itemToString(value) : undefined,`
It can't be undefined, we don't get an error just because of our ts falgs.
If `inputValue` is controllable it must be a string